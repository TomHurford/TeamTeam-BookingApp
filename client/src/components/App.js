import Home from "./Home";
import "../styles/App.css";
import React, {useEffect} from "react";
import Contact from "./Contact";
import Login from './Login';
import Purchase from './Purchase';
import PayPal from './PayPal';
import { Routes, Route } from "react-router-dom";
import Navbar from "./Navbar";
import EventDetails from "./Events/EventDetails";
import Basket from "./Basket";
import ViewSociety from "./Societies/ViewSociety";
import CreateSocietyForm from "./Societies/CreateSocietyForm";
import SearchSocieties from "./Societies/SearchSocieties";
import EditSocietyForm from "./Societies/EditSocietyForm";

//Routes to connect to the homepage, the contact page and other pages which can be added here

function App() {
  const [tickets, setTickets] = React.useState([]);
  const [ticketId, setTicketId] = React.useState(0);
  const [totalPrice, setTotalPrice] = React.useState(0);

  useEffect(() => {
    const storedTickets = sessionStorage.getItem('tickets');
    const storedTicketId = sessionStorage.getItem('ticketId');
    const storedTotalPrice = sessionStorage.getItem('totalPrice');
    if(storedTickets) {
      setTickets(JSON.parse(storedTickets));
      console.log("current tickets");
      console.log(tickets);
      console.log("stored tickets");
      console.log(JSON.parse(storedTickets));
      console.log(storedTickets);

    }
    if(storedTicketId) {
      setTicketId(JSON.parse(storedTicketId));
    }
    if(storedTotalPrice) {
      setTotalPrice(JSON.parse(storedTotalPrice));
    }
  }, []);

  useEffect(() => {
    sessionStorage.setItem('tickets', JSON.stringify(tickets));
    sessionStorage.setItem('ticketId', JSON.stringify(ticketId));
    sessionStorage.setItem('totalPrice', JSON.stringify(totalPrice));
  }, [tickets, ticketId, totalPrice]);

  const addTicket = (ticket) => {
    setTickets([...tickets, { ticket, id: ticketId }]);
    setTicketId(ticketId + 1);
    setTotalPrice(totalPrice + ticket.price);
  };

  const removeTicket = (ticket) => {
    setTickets(tickets.filter((t) => t.id !== ticket.id));
    setTotalPrice(totalPrice - ticket.ticket.price);
  };

  return (
    <div>
      <Navbar />
      <Routes>
        <Route path="/" element={<Home />}></Route>
        <Route path = "/login" element={<Login/>}></Route>
        <Route path="/contact" element={<Contact />}></Route>
        <Route path="/purchase" element={<Purchase />}></Route>
        <Route path="/paypal" element={<PayPal />}></Route>

        <Route
          path="/event-details"
          element={<EventDetails addTicket={addTicket} />}
        ></Route>
        <Route
          path="/basket"
          element={
            <Basket
              tickets={tickets}
              removeTicket={removeTicket}
              totalPrice={totalPrice}
            />
          }
        ></Route>
        
        <Route path="/societies/:id/:name?" element={<ViewSociety />} />
        <Route path="/societies" element={<SearchSocieties />} />
        <Route path="/create-society" element={<CreateSocietyForm />} />
        <Route path="/edit-society" element={<EditSocietyForm />} />
      </Routes>
    </div>
  );
}

export default App;
