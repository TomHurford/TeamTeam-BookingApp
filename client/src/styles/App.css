.App {
  text-align: center;
}

.App-logo {
  height: 40vmin;
  pointer-events: none;
}

@media (prefers-reduced-motion: no-preference) {
  .App-logo {
    animation: App-logo-spin infinite 20s linear;
  }
}

.App-header {
  background-color: #282c34;
  min-height: 100vh;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  font-size: calc(10px + 2vmin);
  color: white;
}

.App-link {
  color: #61dafb;
}

@keyframes App-logo-spin {
  from {
    transform: rotate(0deg);
  }
  to {
    transform: rotate(360deg);
  }
}

/* BELOW IS RANDOM CSS FOR VIEWSOCIETY PAGE **REMOVE LATER** */
* {
  box-sizing: border-box;
}

body {
  font-family: Arial;
  padding: 20px;
  background: #f1f1f1;
}

/* Header/Blog Title */
.header {
  padding: 30px;
  font-size: 40px;
  text-align: center;
  background: white;
}

/* Create two unequal columns that floats next to each other */
/* Left column */
.leftcolumn {
  float: left;
  width: 75%;
}

/* Right column */
.rightcolumn {
  float: left;
  width: 25%;
  padding-left: 20px;
}

/* Fake image */
.fakeimg {
  background-color: #aaa;
  width: 100%;
  padding: 20px;
}

/* Add a card effect for articles */
.card {
  background-color: white;
  padding: 20px;
  margin-top: 20px;
}

/* Clear floats after the columns */
.row:after {
  content: "";
  display: table;
  clear: both;
}

/* Footer */
/* .footer {
  padding: 20px;
  text-align: center;
  background: #ddd;
  margin-top: 20px;
} */

/* Responsive layout - when the screen is less than 800px wide, make the two columns stack on top of each other instead of next to each other */
@media screen and (max-width: 800px) {
  .leftcolumn,
  .rightcolumn {
    width: 100%;
    padding: 0;
  }
}

.container-page {
  position: relative;
  min-height: 100vh;
}

.content{
  padding-bottom: 150px;  
}