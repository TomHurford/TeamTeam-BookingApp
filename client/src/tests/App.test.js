import React from 'react';
import App from '../components/App';
import { BrowserRouter } from "react-router-dom";
import renderer from 'react-test-renderer';
// Snapshot test for App
test('App snapshot test', () => {
  const component = renderer.create(
    <BrowserRouter>
      <App />
    </BrowserRouter>
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});