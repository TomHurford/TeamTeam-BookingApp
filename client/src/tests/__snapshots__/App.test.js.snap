// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App snapshot test 1`] = `
<div>
  <div
    className="nav-container"
  >
    <nav
      className="navbar"
    >
      <ul
        className="navbarElements"
      >
        <a
          href="/"
          onClick={[Function]}
        >
          <li
            className="left"
          >
            <img
              alt="Logo"
              src="logo.png"
            />
          </li>
        </a>
        <a
          href="/"
          onClick={[Function]}
        >
          <li>
            Home
          </li>
        </a>
        <a
          href="/societies"
          onClick={[Function]}
        >
          <li>
            Societies
          </li>
        </a>
        <a
          href="/logout"
          onClick={[Function]}
        >
          <li>
            Logout
          </li>
        </a>
        <a
          href="/profile"
          onClick={[Function]}
        >
          <li>
            Profile
          </li>
        </a>
        <a
          href="/contact"
          onClick={[Function]}
        >
          <li>
            Contact
          </li>
        </a>
        <a
          href="/purchase/future"
          onClick={[Function]}
        >
          <li>
            Purchase
          </li>
        </a>
        <a
          href="/basket"
          onClick={[Function]}
        >
          <li
            data-testid="basket-icon"
          >
            <img
              alt="Basket"
              src="basket.png"
            />
          </li>
        </a>
      </ul>
    </nav>
  </div>
  <div
    className="page-container"
  >
    <div
      className="underlay"
    />
    <div
      className="homePage"
      data-testid="home-component"
    >
      <div
        className="welcome"
        data-testid="welcome-message"
      >
        <h1>
          Welcome to Ticketopia!
        </h1>
      </div>
      <input
        className="searchBar"
        data-testid="search-bar"
        onChange={[Function]}
        onKeyDown={[Function]}
        placeholder="Search for events..."
        type="text"
        value=""
      />
      <div
        className="events"
        data-testid="events-list"
      />
    </div>
  </div>
</div>
`;
